core/src/main/scala/akka/persistence/cassandra/journal/TagWriters.scala

```diff
@@ -184,25 +186,63 @@ import akka.util.ByteString
   }
 
   private def writeTagScanning(): Unit = {
-    val updates = toBeWrittenScanning
+
+    val updates = toBeWrittenScanning.toVector
     // current pendingScanning will be written on next tick, if no write failures
     toBeWrittenScanning = pendingScanning
     // collect new until next tick
     pendingScanning = Map.empty
 
-    if (updates.nonEmpty && log.isDebugEnabled)
-      log.debug("Update tag scanning [{}]", updates.toSeq.mkString(","))
+    if (updates.nonEmpty) {
 
-    tagWriterSession.tagScanningPs.foreach { ps =>
-      updates.foreach {
-        case (pid, seqNr) =>
-          tagWriterSession.executeStatement(ps.bind(pid, seqNr: JLong))
-            .failed.foreach { t =>
-              log.warning("Writing tag scanning failed. Reason {}", t)
-              self ! TagWriteFailed(t)
-            }
+      if (log.isDebugEnabled) {
+        val maxPrint = 20
+        log.debug(
+          "Update tag scanning [{}]",
+          if (updates.size <= maxPrint) updates.take(maxPrint).mkString(",")
+          else updates.take(maxPrint).mkString(",") + s" ...and ${updates.size - 20} more")
+      }
+
+      tagWriterSession.tagScanningPs.foreach { ps =>
+
+        val startTime = System.nanoTime()
+
+        def writeTagScanningBatch(group: Seq[(String, Long)]): Future[Done] = {
+          val batch = new BatchStatement(BatchStatement.Type.UNLOGGED)
+          group.foreach {
+            case (pid, seqNr) => batch.add(ps.bind(pid, seqNr: JLong))
+          }
+          tagWriterSession.executeStatement(batch)
+        }
+
+        // Group the updates into 500 statements per UNLOGGED BatchStatement. These
+        // are executed sequentially to not induce too much load that might influence
+        // performance of other writes and reads. See issue #408.
+        // The size of the data is small and fixed so no need to configure the batch size.
+        val batchIterator: Iterator[Future[Done]] = updates.grouped(500).map(writeTagScanningBatch)
```

Comment:
this seems like quite a large batch size. The default for logging a warning in c* server side is 5kb serialized size which this may hit?

Reply:
Off by 1 0 error :)